src/supervisor.tex

```diff
@@ -1005,8 +1006,11 @@ \subsection{Supervisor Memory-Management Fence Instruction}
 these data structures; however, these implicit references are ordinarily not
 ordered with respect to explicit loads and stores.  Executing
 an SFENCE.VMA instruction guarantees that any previous stores already visible
-to the current RISC-V hart are ordered before all subsequent implicit references
-from that hart to the memory-management data structures.
+to the current RISC-V hart are ordered before certain implicit references by
+subsequent instructions in that hart to the memory-management data structures,
+and vice versa.  SFENCE.VMA is also used to invalidate entries in the
```

Comment:
This is a little bit confusing descriptively, since in previous versions of the document (and possibly elsewhere in the current one, I haven't reread other chapters in the new context) the ordering was intended to subsume incoherent caching.  If the ordering is now separate from the incoherent caching that changes the interpretation of various things.  A more formal model would help.

Reply:
Good feedback.  I added a new paragraph of commentary explaining why we did this: https://github.com/riscv/riscv-isa-manual/pull/611/files#diff-1673d75098c5b339d06598ff6bb126f7a429d6bed61593e9b41092ee0587c9b9R1530.  Hopefully that helps.

And yes, integration with the formal model is on the task group's to-do list, and of course if we find any discrepancies along the way, we'll make sure to update this draft before ratification.

Reply:
That does help, thank you.